service/src/main/java/com/ge/predix/acs/privilege/management/SubjectPrivilegeManagementController.java

```diff
@@ -41,29 +36,55 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.ge.predix.acs.commons.web.BaseRestApi;
-import com.ge.predix.acs.commons.web.RestApiException;
-import com.ge.predix.acs.commons.web.UriTemplateUtils;
-import com.ge.predix.acs.rest.BaseSubject;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 
-import io.swagger.annotations.ApiOperation;
-import io.swagger.annotations.ApiResponse;
-import io.swagger.annotations.ApiResponses;
+import static com.ge.predix.acs.commons.web.AcsApiUriTemplates.SUBJECTS_URL;
+import static com.ge.predix.acs.commons.web.AcsApiUriTemplates.SUBJECT_URL;
+import static com.ge.predix.acs.commons.web.AcsApiUriTemplates.V1;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.created;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.noContent;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.notFound;
+import static com.ge.predix.acs.commons.web.ResponseEntityBuilder.ok;
+import static org.springframework.web.bind.annotation.RequestMethod.DELETE;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+import static org.springframework.web.bind.annotation.RequestMethod.PUT;
 
 @RestController
 public class SubjectPrivilegeManagementController extends BaseRestApi {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubjectPrivilegeManagementController.class);
+
     @Autowired
     private PrivilegeManagementService service;
 
+    private void failIfParentsSpecified(final List<BaseSubject> subjects) {
+        for (BaseSubject subject : subjects) {
+            if (!CollectionUtils.isEmpty(subject.getParents())
+                && !Arrays.asList(this.getEnvironment().getActiveProfiles()).contains("titan")) {
+                throw new RestApiException(HttpStatus.NOT_IMPLEMENTED, PARENTS_ATTR_NOT_SUPPORTED_MSG);
```

Comment:
The check appears to be independent of subjects we are iterating in for loop, can we take this if statement !Arrays.asList(this.getEnvironment().getActiveProfiles()).contains("titan")) out of the for loop and additional code outside the for loop so that logic is still same.